src/wazuh_modules/wm_oscap.c

```diff
@@ -341,6 +346,39 @@ cJSON *wm_oscap_dump(const wm_oscap *oscap) {
 
 // Destroy data
 
+#ifdef WIN32
+DWORD WINAPI wm_oscap_destroy(void *oscap) {
+    wm_oscap *oscap_ptr = (wm_oscap *)oscap;
+    wm_oscap_eval *cur_eval;
+    wm_oscap_eval *next_eval;
+    wm_oscap_profile *cur_profile;
+    wm_oscap_profile *next_profile;
+
+    // Delete evals
+
+    for (cur_eval = oscap_ptr->evals; cur_eval; cur_eval = next_eval) {
+
+        // Delete profiles
+
+        for (cur_profile = cur_eval->profiles; cur_profile; cur_profile = next_profile) {
+            next_profile = cur_profile->next;
+            free(cur_profile->name);
+            free(cur_profile);
+        }
+
+        next_eval = cur_eval->next;
+        free(cur_eval->path);
+        free(cur_eval->xccdf_id);
+        free(cur_eval->oval_id);
+        free(cur_eval->ds_id);
+        free(cur_eval->cpe);
+        free(cur_eval);
+    }
+
+    free(oscap_ptr);
+    return 0;
```

Comment:
this could be summarized with a #ifdef WIN32 on the return to avoid the entire duplication of the function.